### PR TITLE
INT-4438 - Zendesk Pagination Error

### PR DESCRIPTION
**Pull request by eXtremeX**
**Root Cause Analysis**
- `nextPage` was used to access next page link instead of the appropriate `next_page` property.
- `next_page` logic was also flawed. API returns a complete link, yet it is being used as a page number (in the query string) in this context, which will cause it to fail right after the first paginated request.
- The reason why a maximum of 100 objects are returned is because no entity limit per page was set. API endpoints have a default upper limit of [100 records per page](https://developer.zendesk.com/api-reference/ticketing/introduction/#using-offset-pagination).
- This is true for [users][users-endpoint], [groups][groups-endpoint], [organizations][organizations-endpoint], and [tickets][tickets-endpoint].

**Proof**
Using the following request and setting maximum records per page to 1
```javascript
const response = await this.apiRequestWithErrorHandling(
    `${path}.json?page=${page}&per_page=1`,
);
```
Results to the following graph
image
The integration will not be able to retrieve records after page 1.

**Solution**
- Zendesk recommends to switch to [cursor pagination](https://developer.zendesk.com/api-reference/ticketing/introduction/#using-cursor-pagination) which is supported by existing endpoints of the application. Logic was updated to support proper cursor pagination.
- Set `paginateEntitiesPerPage` to 50

[users-endpoint]: https://developer.zendesk.com/api-reference/ticketing/users/users/#pagination
[groups-endpoint]: https://developer.zendesk.com/api-reference/ticketing/groups/groups/#pagination
[organizations-endpoint]: https://developer.zendesk.com/api-reference/ticketing/organizations/organizations/#pagination
[tickets-endpoint]: https://developer.zendesk.com/api-reference/ticketing/tickets/tickets/#pagination

**Proof**
`paginateEntitiesPerPage = 1;`
image

`paginateEntitiesPerPage = 50;`
image

Since both returned the same total entities. This means pagination is working as intended.

### CODE REVIEW

File: .github/workflows/gitleaks.yml [hunk 1]
```diff
@@ -11,4 +11,4 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - name: gitleaks-action
-        uses: zricethezav/gitleaks-action@master
```
Comment (VDubber) [inline .github/workflows/gitleaks.yml:14]: Very recently we discovered that GitHub Enterprise tooling takes care of commit secret prevention. We will no longer be using gitleaks. Feel to remove this file and `.gitleaks.yml`

**Review by austinkelleher — APPROVED** [review]
Thank you!

File: .github/workflows/gitleaks.yml (hunk 1 above)
Comment (austinkelleher) [reply .github/workflows/gitleaks.yml:14]: @eXtremeX Take a look at this: https://github.com/JupiterOne/integration-template/pull/96/files

Feel free to address in a separate PR if you'd like

Comment (eXtremeX) [reply .github/workflows/gitleaks.yml:14]: Thanks for letting me know that @VDubber!

**Pull request merged**